gtk4/src/subclass/widget.rs

```diff
@@ -955,16 +954,37 @@ pub unsafe trait WidgetClassSubclassExt: ClassStruct {
         }
     }
 
-    #[doc(alias = "gtk_widget_class_install_action")]
-    fn install_action<
-        F: Fn(&<<Self as ClassStruct>::Type as ObjectSubclass>::Type, &str, Option<&Variant>)
-            + 'static,
-    >(
+    fn install_action_async<Fut, F>(
         &mut self,
         action_name: &str,
         parameter_type: Option<&str>,
         activate: F,
-    ) {
+    ) where
+        F: Fn(<<Self as ClassStruct>::Type as ObjectSubclass>::Type, &str, Option<&Variant>) -> Fut
```

Comment:
I couldn't mange to this this to work with `&self` here

Reply:
You mean `&imp` vs. `wrapper`?

Reply:
the `install_action` variant takes `&Self::Type` as a first param of the callback where the `install_action_async`, takes `Self::Type`. I wonder if I should make them both take `Self::Type`

Reply:
Ah so the question is just about the reference. `&T` seems more consistent

Reply:
Yes but that fails with 
```
error: lifetime may not live long enough
  --> examples/video_player/video_player_window/imp.rs:49:49
   |
49 |               |win, _action_name, _action_target| async move {
   |  ______________---______________________________-_^
   | |              |                                |
   | |              |                                return type of closure `impl Future<Output = ()>` contains a lifetime `'2`
   | |              has type `&'1 video_player_window::VideoPlayerWindow`
50 | |                 let dialog = &win.imp().dialog;
51 | |                 dialog.set_transient_for(Some(win));
52 | |                 if dialog.run_future().await == gtk::ResponseType::Accept {
...  |
55 | |                 dialog.destroy();
56 | |             },
   | |_____________^ returning this value requires that `'1` must outlive `'2`

```
when trying to make use of it

Reply:
You need to move a `T` into the `async` block in either case

Reply:
yes, which is something I don't have to care about if I use `T` directly instead of `&T`

Reply:
otherwise you will end up using yet another clone! block and avoiding it is what motivated this PR

Reply:
Oh I see... because we don't have async closures yet. Nevermind then :)